ibodata/lateral_profile.py

```diff
@@ -23,48 +23,79 @@ def penumbra_left(self):
     def field_ratio(self, level):
         """
         In case of good data returns positive number
+        Level has to be >= 0.0 and <= 1 or exception is raised
         In case of corrupted data returns nan
         """
+        if level < 0.0 or level > 1.0:
+            raise ValueError("Expected level to be between 0.0 and 1.0")
         return self.width(level) / self.width(0.5)
 
     def symmetry(self, level):
         """
         In case of good data returns positive number
+        Level has to be >= 0 and <= 1 or exception is raised
         In case of corrupted data returns nan
         """
+        if level < 0 or level > 1.0:
+            raise ValueError("Expected level to be between 0 and 1")
         a = math.fabs(self.x_at_y(level, False))
         b = math.fabs(self.x_at_y(level, True))
         return (math.fabs(a - b) / (a + b)) * 200.0
 
     def flatness_50(self):
+        """
+        In case of corrupted data returns nan
+        Returns value between points if max/min value occurs on border
+        """
         d = (self.penumbra_left() + self.penumbra_right()) / 2
-        if d == np.nan:
+        if d != d:
             return np.nan
         left = self.x_at_y(0.5) + 2.0 * d
         right = self.x_at_y(0.5, True) - 2.0 * d
 
-        p_max = np.max(self.y[np.logical_and(self.x <= right, self.x >= left)])
-        p_min = np.min(self.y[np.logical_and(self.x <= right, self.x >= left)])
+        p_max = np.max(
+            np.append(self.y[np.logical_and(self.x <= right, self.x >= left)], [self.y_at_x(right), self.y_at_x(left)]))
+        p_min = np.min(
+            np.append(self.y[np.logical_and(self.x <= right, self.x >= left)], [self.y_at_x(right), self.y_at_x(left)]))
 
         return ((p_max - p_min) / (p_max + p_min)) * 100.0
 
     def flatness_90(self):
+        """
+        In case of corrupted data returns nan
+        Returns value between points if max/min value occurs on border
+        """
         d = (self.penumbra_left() + self.penumbra_right()) / 2
-        if d == np.nan:
+        if d != d:
             return np.nan
         left = self.x_at_y(0.9) + d
         right = self.x_at_y(0.9, True) - d
 
-        p_max = np.max(self.y[np.logical_and(self.x <= right, self.x >= left)])
-        p_min = np.min(self.y[np.logical_and(self.x <= right, self.x >= left)])
+        p_max = np.max(
+            np.append(self.y[np.logical_and(self.x <= right, self.x >= left)], [self.y_at_x(right), self.y_at_x(left)]))
+        p_min = np.min(
+            np.append(self.y[np.logical_and(self.x <= right, self.x >= left)], [self.y_at_x(right), self.y_at_x(left)]))
 
         return ((p_max - p_min) / (p_max + p_min)) * 100.0
 
     def asymmetry(self):
+        """
+        In case of corrupted data returns nan
```

Comment:
Does it return nan or np.nan ?

Reply:
Doesn't matter, because type(np.nan) is equal type(float('nan'))